backend/src/main/kotlin/no/nav/emottak/application/api/RegistrerMeldingerApi.kt

```diff
@@ -1,225 +1,301 @@
-                    returnBadRequest("Mangler parameter: partnerId")
-                    return@get
-                }
-                log.info("Henter info for partnerid : $partnerid")
-                val partnerIdInfo = meldingService.partnerid(partnerid)
-                log.info("Partner info for $partnerid: ${partnerIdInfo.size}")
-                call.respond(partnerIdInfo)
-            }
+// Henting av ulike from_role, service, action for bruk som filter på EBMS-hendelser
+@InternalAPI
+fun Route.hentRollerServicesAction(httpClient: HttpClient): Route =
+    get("/hentrollerservicesaction") {
+        val url = "$eventManagerUrl/filter-values"
+        log.info("Henter filter-verdier for rolle, service, action ($url)")
+        executeREST(httpClient, url)
+    }
 
-            // Feilstatistikk (frontend: /feilstatistikk)
-            get("/hentfeilstatistikk") {
-                val (fom, tom) = localDateTimeLocalDateTimePair() ?: return@get
-                log.info("Kjører dabasespørring for å hente feil statistikk...")
-                val feilStatistikk = meldingService.feilstatistikk(fom, tom)
-                log.info("feil statistikk antall : ${feilStatistikk.size}")
-                call.respond(feilStatistikk)
-            }
+// Henting av sist brukt-datoer fra gamle og nye emottak (/lastused)
+@InternalAPI
+fun Route.hentSistBrukt(
+    meldingService: MessageQueryService,
+    httpClient: HttpClient,
+): Route =
+    get("/hentsistbrukt") {
+        log.info("Henter sist brukt-timestamps")
+
+        // Gamle emottak:
+        val response = hentSistBruktGamleEmottak(meldingService).toMutableList()
 
-            // Henting av ulike from_role, service, action for bruk som filter på EBMS-hendelser
-            get("/hentrollerservicesaction") {
-                val url = "$eventManagerUrl/filter-values"
-                log.info("Henter filter-verdier for rolle, service, action ($url)")
-                executeREST(url)
+        // Nye emottak:
+        val responseEbms = hentSistBruktNyeEmottak(httpClient, response) ?: return@get
+        responseEbms.forEach { (cpaId, lastUsed) ->
+            val lastUsedDate = lastUsed?.split("T")[0]
+            response
+                .find { cpaLastUsed ->
+                    cpaId == cpaLastUsed.cpaId
+                }?.let {
+                    it.lastUsedEbms = lastUsedDate
+                } ?: run {
+                response.add(CpaLastUsed(cpaId, null, lastUsedDate))
             }
         }
+        call.respond(response)
+    }
+
+private fun hentSistBruktGamleEmottak(meldingService: MessageQueryService): List<CpaLastUsed> {
+    log.info("Henter sist brukt-timestamps fra gamle emottak")
+    return meldingService.sistBrukt().also {
+        log.info("Antall CPA sist brukt gamle emottak: ${it.size}")
+    }
+}
+
+@InternalAPI
+private suspend fun RoutingContext.hentSistBruktNyeEmottak(
+    httpClient: HttpClient,
+    responseFraGamleEmottak: MutableList<CpaLastUsed>,
```

Comment:
Kan `responseFraGamleEmottak` kanskje flyttes ut av `hentSistBruktNyeEmottak`? Altså at den fungerer likt som `hentSistBruktGamleEmottak`, og returnerer `List<CpaLastUsed>`. Selve flettingen av de to listene kan da foregå i `Route.hentSistBrukt`. Kan fortsatt få til den PartialContent returneringen når `hentSistBruktNyeEmottak` returnerer `null`, om man ønsker det.